Cython/Compiler/Code.py

```diff
@@ -1619,44 +1629,41 @@ def generate_string_constants(self):
                 else:
                     encoding = '"%s"' % py_string.encoding.lower()
 
-                self.parts['module_state'].putln("PyObject *%s;" % py_string.cname)
-                self.parts['module_state_defines'].putln("#define %s %s->%s" % (
+                self.parts['module_state_defines'].putln("#define %s %s->%s[%s]" % (
                     py_string.cname,
                     Naming.modulestateglobal_cname,
-                    py_string.cname))
-                self.parts['module_state_clear'].putln("Py_CLEAR(clear_module_state->%s);" %
-                    py_string.cname)
-                self.parts['module_state_traverse'].putln("Py_VISIT(traverse_module_state->%s);" %
-                    py_string.cname)
+                    Naming.stringtab_cname,
+                    n))
                 if py_string.py3str_cstring:
                     w.putln("#if PY_MAJOR_VERSION >= 3")
-                    w.putln("{&%s, %s, sizeof(%s), %s, %d, %d, %d}," % (
-                        py_string.cname,
+                    w.putln("{%s, sizeof(%s), %s, %d, %d, %d}, /* PyObject cname: %s */" % (
                         py_string.py3str_cstring.cname,
                         py_string.py3str_cstring.cname,
                         '0', 1, 0,
-                        py_string.intern
+                        py_string.intern,
+                        py_string.cname
                         ))
                     w.putln("#else")
-                w.putln("{&%s, %s, sizeof(%s), %s, %d, %d, %d}," % (
-                    py_string.cname,
+                w.putln("{%s, sizeof(%s), %s, %d, %d, %d}, /* PyObject cname: %s */" % (
```

Comment:
I've added comments with the PyObject cname because I think it's useful to be able to trace this when inspecting the code.